app/src/main/java/org/apache/fineract/ui/online/DashboardActivity.java

```diff
@@ -74,6 +76,8 @@ protected void onCreate(Bundle savedInstanceState) {
             StartSyncJob.scheduleItNow();
             preferencesHelper.setFetching(false);
         }
+
+        registerNetworkRegister();
```

Comment:
I think there is no need to register Broadcast Reciever dynamically again if you have already registered in Manifest. It is redundant, so please remove it.

Reply:
Actually, if we do not register it in code, then it will be registered everywhere in the app. And even for that, we would need to create a separate process to handle the receiver. We only need it work when the user has logged in.
Also, this is the best and most non expensive way to implement this. I have thoroughly tested it and it is working as intended.

Reply:
Yeah, I totally agree with you. But you have registered the broadcast in Manifest file too. If you think the dynamic registration is better then there is no need to register in the Manifest file. Right now you are registering one receiver in two places.

Reply:
This is just to avoid any gaps. If the reviewers feel the same, I will change it. :)